app/models/manageiq/providers/foreman/configuration_manager.rb

```diff
@@ -13,10 +13,16 @@ class ManageIQ::Providers::Foreman::ConfigurationManager < ManageIQ::Providers::
            :authentications,
            :connect,
            :endpoints,
+           :url,
+           :url=,
            :verify_credentials,
            :with_provider_connection,
            :to => :provider
 
+  belongs_to :provider, :autosave => true, :dependent => :destroy
+
+  before_save :ensure_provider_name_and_zone
```

Comment:
The `before_save` doesn't really deal with the new `name` and `zone` value coming from an edit action. I debugged the method below and the `name` value was the one stored in the DB and not the one I'm changing the value to. It works well on create, but not when editing through the API.